common/src/main/scala/no/ndla/common/model/api/StringBasedValue.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Part of NDLA common
+ * Copyright (C) 2025 NDLA
+ *
+ * See LICENSE
+ *
+ */
+
+package no.ndla.common.model.api
+
+import sttp.tapir.{Codec, CodecFormat, DecodeResult, Schema}
+import io.circe.{Decoder, Encoder, HCursor}
+import io.circe.syntax.*
+
+abstract class StringBasedValue(val value: String) extends Comparable[StringBasedValue] {
+  override def equals(obj: Any): Boolean = {
+    obj match {
+      case that: StringBasedValue => this.value == that.value
+      case _                      => false
+    }
+  }
+  override def hashCode(): Int                         = value.hashCode()
+  override def compareTo(other: StringBasedValue): Int = this.value.compareTo(other.value)
+  override def toString: String                        = value
+}
+
+object StringBasedValue {
+  implicit def schema[T <: StringBasedValue]: Schema[T] = Schema.string.asInstanceOf[Schema[T]]
```

Comment:
Trenger vi denne casten?
```suggestion
  implicit def schema[T <: StringBasedValue]: Schema[T] = Schema.string
```

Reply:
Sikkert ikkje. Har berre gjenbrukt koden fra LanguageCode. kan endre.